### PR TITLE
add 'User-Agent'-header to pooch.retrieve

### LINKED ISSUE #9774: docs builds are broken
### What happened?

```

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/xray/conda/9766/lib/python3.12/site-packages/nbsphinx/__init__.py", line 634, in parse
    rststring, resources = exporter.from_notebook_node(nb, resources)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/docs/checkouts/readthedocs.org/user_builds/xray/conda/9766/lib/python3.12/site-packages/nbsphinx/__init__.py", line 397, in from_notebook_node
    nb, resources = pp.preprocess(nb, resources)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/docs/checkouts/readthedocs.org/user_builds/xray/conda/9766/lib/python3.12/site-packages/nbconvert/preprocessors/execute.py", line 103, in preprocess
    self.preprocess_cell(cell, resources, index)
  File "/home/docs/checkouts/readthedocs.org/user_builds/xray/conda/9766/lib/python3.12/site-packages/nbconvert/preprocessors/execute.py", line 124, in preprocess_cell
    cell = self.execute_cell(cell, index, store_history=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/docs/checkouts/readthedocs.org/user_builds/xray/conda/9766/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/docs/checkouts/readthedocs.org/user_builds/xray/conda/9766/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/docs/checkouts/readthedocs.org/user_builds/xray/conda/9766/lib/python3.12/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/docs/checkouts/readthedocs.org/user_builds/xray/conda/9766/lib/python3.12/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
ds = xr.tutorial.load_dataset("era5-2mt-2019-03-uk.grib", engine="cfgrib")
------------------


[0;31mHTTPError[0m: 403 Client Error: Forbidden for url: https://raw.githubusercontent.com/pydata/xarray-data/master/era5-2mt-2019-03-uk.grib

```

### What did you expect to happen?

_No response_

### Minimal Complete Verifiable Example

_No response_

### MVCE confirmation

- [ ] Minimal example — the example is as focused as reasonably possible to demonstrate the underlying issue in xarray.
- [ ] Complete example — the example is self-contained, including all data and the text of any traceback.
- [ ] Verifiable example — the example copy & pastes into an IPython prompt or [Binder notebook](https://mybinder.org/v2/gh/pydata/xarray/main?urlpath=lab/tree/doc/examples/blank_template.ipynb), returning the result.
- [ ] New issue — a search of GitHub Issues suggests this is not a duplicate.
- [ ] Recent environment — the issue occurs with the latest version of xarray and its dependencies.

### Relevant log output

_No response_

### Anything else we need to know?

_No response_

### Environment

<details>

</details>

### PR DESCRIPTION
**Pull request by kmuehlbauer**
- [X] Closes #9774
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

File: xarray/tutorial.py [hunk 1]
```diff
@@ -157,8 +158,16 @@ def open_dataset(
 
         url = f"{base_url}/raw/{version}/{path.name}"
 
+    def _xarray_downloader(url, output_file, xrpooch):
+        """Create Downloader which adds request-headers"""
+        headers = {"User-Agent": f"xarray {sys.modules['xarray'].__version__}"}
+        https = pooch.HTTPDownloader(headers=headers)
+        https(url, output_file, xrpooch)
```
Comment (mathause) [inline xarray/tutorial.py:165]:
```suggestion
    
    headers = {"User-Agent": f"xarray {sys.modules['xarray'].__version__}"}
    downloader = pooch.HTTPDownloader(headers=headers)
```

File: xarray/tutorial.py [hunk 2]
```diff
@@ -157,8 +158,16 @@ def open_dataset(
...
+
     # retrieve the file
-    filepath = pooch.retrieve(url=url, known_hash=None, path=cache_dir)
+    filepath = pooch.retrieve(
+        url=url, known_hash=None, path=cache_dir, downloader=_xarray_downloader
```
Comment (mathause) [inline xarray/tutorial.py:169]:
```suggestion
        url=url, known_hash=None, path=cache_dir, downloader=downloader
```

**Review by mathause — COMMENTED** [review]
Can probably just leave away the version from the string... & is there a reason you don't pass the downloader directly?

**Comment by kmuehlbauer** [comment]
> & is there a reason you don't pass the downloader directly?

No, I was just somehow following the pooch tutorial :grin:

**Comment by kmuehlbauer** [comment]
> Can probably just leave away the version from the string.

It might be useful to know the actual version for debugging, but I have no strong opinion.

**Comment by mathause** [comment]
Yes that's also good for me (just thought it's not worth debugging forever)

**Comment by kmuehlbauer** [comment]
> Yes that's also good for me (just thought it's not worth debugging forever)

I was more thinking about situations where users experience these issues again. It might be that one time `User-Agent` is not enough to add to the headers.

**Comment by kmuehlbauer** [comment]
@mathause Thanks for the review. I think this is ready to go.

**Review by mathause — APPROVED** [review]
Thanks for the fast PR!

**Comment by keewis** [comment]
for reference, `python-requests` (which `pooch` uses to do the download) adds a `User-Agent` of `python-requests/<version>`. My guess is that the combination of RTD servers + `python-requests` user agent was blocked, which is why changing to something else helps.

**Pull request merged**